tests/tt_eager/python_api_testing/unit_testing/test_sharded_tensor.py

```diff
@@ -0,0 +1,215 @@
+            if not passing:
+                print("Tile " + str(tile_idx) + " not_matching ")
+            col_idx = col_idx + 1
+        row_idx = row_idx + 1
+
+
+@pytest.mark.parametrize(
+    "tt_dtype",
+    [
+        ttl.tensor.DataType.UINT32,
+        ttl.tensor.DataType.BFLOAT16,
+    ],
+)
+@pytest.mark.parametrize(
+    "tensor_shape, shard_scheme, shard_shape, compute_grid",
+    [
+        ([1, 4, 64, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (64, 64), (0, 3)),
+        ([1, 1, 128, 128], ttl.tensor.TensorMemoryLayout.BLOCK_SHARDED, (64, 64), (0, 3)),
+        ([1, 1, 2048, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (512, 64), None),
+        ([1, 1, 2048, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (1024, 64), None),
+        ([1, 1, 4096, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (1024, 64), None),
+        ([1, 1, 8192, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (1024, 64), None),
+        ([1, 1, 14336, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (1024, 64), None),
+        ([1, 1, 100352, 64], ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED, (1024, 64), None),
+        ([1, 1, 256, 32], ttl.tensor.TensorMemoryLayout.WIDTH_SHARDED, (32, 32), None),
+        ([1, 1, 128, 64], ttl.tensor.TensorMemoryLayout.WIDTH_SHARDED, (32, 64), None),
+        ([1, 1, 100352, 64], ttl.tensor.TensorMemoryLayout.BLOCK_SHARDED, (2048, 32), None),
+        ([1, 1, 2048, 64], ttl.tensor.TensorMemoryLayout.BLOCK_SHARDED, (1024, 64), None),
+    ],
+)
+@pytest.mark.parametrize(
+    "shard_orientation",
+    [ttl.tensor.ShardOrientation.ROW_MAJOR, ttl.tensor.ShardOrientation.COL_MAJOR],
+)
+def test_tensor_conversion_between_torch_and_tt_tile(
+    tt_dtype, device, tensor_shape, shard_scheme, shard_shape, compute_grid, shard_orientation
+):
+    dtype = tt_dtype_to_torch_dtype[tt_dtype]
+    if compute_grid == None:
+        compute_grid = ttl.tensor.CoreCoord(
+            device.compute_with_storage_grid_size().x - 1, device.compute_with_storage_grid_size().y - 1
+        )
+    else:
+        compute_grid = ttl.tensor.CoreCoord(compute_grid[0], compute_grid[1])
+
+    shard_grid = ttl.tensor.CoreRangeSet({ttl.tensor.CoreRange(ttl.tensor.CoreCoord(0, 0), compute_grid)})
+    shard_halo = False
+    shard_spec = ttl.tensor.ShardSpec(shard_grid, shard_shape, shard_orientation, shard_halo)
+
+    two_d_shape = (tensor_shape[0] * tensor_shape[1] * tensor_shape[2], tensor_shape[3])
+    num_tiles_width = (two_d_shape[1]) / TILE_WIDTH
+    num_tiles_height = (two_d_shape[0]) / TILE_HEIGHT
+    num_shards = (two_d_shape[0] / shard_shape[0]) * (two_d_shape[1] / shard_shape[1])
+
+    torch_tensor = None
+    for row_idx in range(0, int(num_tiles_height)):
+        tile_row = None
+        for col_idx in range(0, int(num_tiles_width)):
+            tile_idx = col_idx + num_tiles_width * row_idx
+            tile = torch.full((1, 1, TILE_WIDTH, TILE_HEIGHT), tile_idx + 1, dtype=dtype)
```

Comment:
For testing, can we use a randint? The tile_idx and page_idx can be kept around for debug.

Reply:
is the issue the readability of the python code?

If so would putting this behind a function help?
The sorted pages I don't think take away anything in terms of testing.

Reply:
I added a debug flag and if its debug I will use the old way to make the tensors, else use randint/rand.

it's in this commit. I'll squash this commit once you have a look at it.
https://github.com/tenstorrent-metal/tt-metal/pull/3790/commits/2d8121597fe847e5e22f17e625107b898c304545

Reply:
looks good!